test/suite_implementation.js

```diff
@@ -65,29 +67,26 @@ module.exports = function(style, options, callback) {
 };
 
 function applyOperations(map, operations, callback) {
-    var operation = operations && operations[0];
-
-    if (!operations || operations.length === 0) {
-        map.once('render', function onRender() {
-            if (map.loaded()) {
-                callback();
-            } else {
-                map.once('render', onRender);
-            }
-        });
-
-    } else if (operation[0] === 'wait') {
-        map.once(operation[1], function() {
-            applyOperations(map, operations.slice(1), callback);
-        });
-
-    } else {
-        try {
+    try {
+        var operation = operations && operations[0];
+        if (!operations || operations.length === 0) {
+            callback();
+
+        } else if (operation[0] === 'wait') {
+            var wait = function() {
+                if (map.loaded()) {
+                    applyOperations(map, operations.slice(1), callback);
+                } else {
+                    map.once('render', wait);
+                }
+            };
+            wait();
+
+        } else {
             map[operation[0]].apply(map, operation.slice(1));
-        } catch (e) {
-            console.error(e.stack);
-            throw e;
+            applyOperations(map, operations.slice(1), callback);
         }
-        applyOperations(map, operations.slice(1), callback);
+    } catch (e) {
```

Comment:
It gets caught by the [exception handler in map](https://github.com/mapbox/mapbox-gl-js/blob/cb814378a30c634dce942dcfbd31f15d8e538b2d/js/ui/map.js#L908-L908), emitted as an `error` event, and logged by the [default handler](https://github.com/mapbox/mapbox-gl-js/blob/cb814378a30c634dce942dcfbd31f15d8e538b2d/js/ui/map.js#L946). I guess this is a difference in how node and a browser handle `console.log(error)` -- the browser behavior of logging the stack trace being much more useful.

What about having the test-suite harness bind an `error` handler that does more complete logging? This would be helpful in general for any time the test-suite happens to cause an error in gl-js internals.